components/eventing-controller/controllers/subscription/beb/reconciler.go

```diff
@@ -245,7 +245,7 @@ func (r *Reconciler) handleDeleteSubscription(ctx context.Context, subscription
 
 	// update condition in subscription status
 	condition := eventingv1alpha1.MakeCondition(eventingv1alpha1.ConditionSubscribed, eventingv1alpha1.ConditionReasonSubscriptionDeleted, corev1.ConditionFalse, "")
-	r.replaceStatusCondition(subscription, condition)
+	_ = r.replaceStatusCondition(subscription, condition)
```

Comment:
Why do we ignore the returned value? 

I suggest to check if we really need to return anything from the `replaceStatusCondition` func, and maybe change its signature to not return anything if it is always ignored.